lib/ps.js

```diff
@@ -0,0 +1,80 @@
+var exec = require('child_process').exec
+var os = require('os')
+var PLATFORM = os.platform()
+
+/**
+  * Get pid informations through ps command
+  * @param   {Number|Number[]}  pid
+  * @param   {Object}           options
+  * @return  {Function}         callback(err, stat)
+  * on os x skip headers with pcpu=,rss=
+  * on linux it could be --no-header
+  * on solaris 11 can't figure out a way to do this properly so...
+  */
+function ps (pid, options, done) {
+  var advanced = options.advanced
+  if (!Array.isArray(pid)) {
+    pid = parseInt(pid, 10)
+  } else {
+    advanced = true
+    pid = pid.map(function (pid) {
+      return parseInt(pid, 10)
+    }).join(',')
+  }
+
+  var cmd = 'ps -o pcpu,rss,pid -p '
+
+  if (advanced) {
+    cmd = 'ps -o pcpu,rss,pid,time -p '
+
+    if (PLATFORM === 'aix') {
+      cmd = 'ps -o pcpu,rssize,pid,time -p '
+    }
+  } else if (PLATFORM === 'aix') {
+    cmd = 'ps -o pcpu,rssize,pid -p ' // this one could work on other platforms see AIX section in man ps
+  }
+
+  exec(cmd + pid, function (err, stdout, stderr) {
```

Comment:
I believe that by using `spawn` instead of `exec` we can achieve less overhead.

While creating tests for:
https://github.com/indexzero/ps-tree/pull/26
I've noticed that `exec` on both windows and linux creates a new shell then from that shell it executes the given command command. So you have 2 new processes created.

Instead spawn seems only to run the provided binary.

Reply:
But things get a bit more complicated since using spawn you will have stdout as stream.
I don't know if It's worth it